### PR TITLE
fetching of the matchesList

### PR DESCRIPTION
**Pull request by xiaozdezhu**

### CODE REVIEW

**Comment by xiaozdezhu** [comment]
- Implement listenToMyMatchesList function to listen on one test user's matches.
- Implement getMatchLists to get all a list of match infos by the matches id.
- Implement getMatchDetail to get correspoding match detail back.

**Comment by xiaozdezhu** [comment]
I don't know how to test for this part. I try to add new match in the test file and call the listenToMyMatchesList function I implemented in the firebase.ts. However the change should be reflected on UI or state change which I don't see how to verify. I'll ask professor help on tomorrow class.

File: src/services/firebase.ts [hunk 1]
```diff
@@ -3,7 +3,9 @@ import { store } from '../stores/index';
 import * as firebase from 'firebase';
 
 // All interactions with firebase must be in this module.
-export module ourFirebase {
+export namespace ourFirebase {
+  let testUser = '518vWpI2HzZajYERnlOvZWpDqQ92';
```
Comment (yoav-zibin) [inline src/services/firebase.ts]: There shouldn't be any test code in this file.

File: src/services/firebase.ts [hunk 2]
```diff
@@ -41,14 +46,43 @@ export module ourFirebase {
...
+    let updateMatchList: any = new Array();
+    snap.forEach(function(childSnapshot: firebase.database.DataSnapshot) {
+      let matchId = childSnapshot.key;
+      updateMatchList.push(getMatchDetail(matchId));
+      return false;
+    });
+    store.dispatch(updateMatchList);
+  }
+
+  function getMatchDetail(matchId: any) {
```
Comment (yoav-zibin) [inline src/services/firebase.ts]: Add types, this will help you find mistakes, e.g., you return a promise (not MatchInfo) from here.

**Pull request merged**